File: greedy/2139_minimum_moves_to_reach_target_score/2139_minimum_moves_to_reach_target_score.py
class Solution:
    def minMoves(self, target: int, maxDoubles: int) -> int:
        if maxDoubles == 0 :
            return target - 1

        result = 0
        while target >1 :
            if maxDoubles > 0:
                if target % 2 == 0:
                    result += 1
                    maxDoubles -= 1
                    target = target //2
                else:
                    result += 1
                    target -= 1
            else:
                result += target - 1
                target = 1
        return result
                    

File: greedy/2139_minimum_moves_to_reach_target_score/2139_minimum_moves_to_reach_target_score.go
package main

func minMoves(target int, maxDoubles int) int {
	result := 0

	for target > 1 {
		if maxDoubles > 0 {
			result += 1
			if target%2 == 0 {
				maxDoubles -= 1
				target = target / 2
			} else {
				target -= 1
			}
		} else {
			result += (target - 1)
			target = 1
		}
	}
	return result
}
